src/test/java/cc/robotdreams/UI/TaskManagementUITest.java

```diff
@@ -0,0 +1,209 @@
+        List<HashMap> allTasks = KanboardAPI.getAllTasks(projectID, 0, 123);
+        List<String> allTasksTitles = new ArrayList<>();
+        for (HashMap entry : allTasks) {
+            allTasksTitles.add(entry.get("title").toString());
+        }
+        logger.debug("allTasksTitles: " + allTasksTitles);
+        Assert.assertListContainsObject(allTasksTitles, taskTitleToAdd, "Title: " + taskTitleToAdd);
+
+        //Teardown
+        testTearDown(projectID, userID);
+    }
+
+    @Test(groups = {"commentTaskWithUI"}, priority = 3)
+    public void addComment() {
+        //-------------------------------------
+        String taskTitleToAdd = "test_task" + RandomStringUtils.random(3, true, true);
+        String comment = "Some text to comment";
+        //------------------------------------
+
+        //Setup
+        HashMap<String,Integer> setupResult = testSetup(usernameToCreate, passwordToCreate, testPrjName);
+        Integer userID = setupResult.get("user_id");
+        Integer projectID = setupResult.get("project_id");
+
+        LoginPage loginPage = new LoginPage();
+        DashboardPage dashboardPage = loginPage.login(usernameToCreate, passwordToCreate);
+        ProjectBoard projectBoard = dashboardPage.goToProjectBoard(projectID);
+        projectBoard.confirm();
+        projectBoard.addTaskToBacklog(taskTitleToAdd);
+        Integer taskID = KanboardAPI.getTaskIDByName(taskTitleToAdd, projectID);
+
+        //Test
+        projectBoard.openContextMenu();
+        projectBoard.addComment(comment);
+        Wait.sleep(500);
+        TaskPage taskPage = projectBoard.goToTaskPage(taskID);
+        taskPage.confirm();
+
+        Assert.assertEquals(comment, taskPage.getComment());
+
+        //Teardown
+        testTearDown(projectID, userID);
+    }
+
+    @Test(groups = {"commentTaskWithUI"}, priority = 4)
+    public void addEmptyComment() {
+        //-------------------------------------
+        String taskTitleToAdd = "test_task" + RandomStringUtils.random(3, true, true);
+        String comment = "";
+        //------------------------------------
+
+        //Setup
+        HashMap<String,Integer> setupResult = testSetup(usernameToCreate, passwordToCreate, testPrjName);
+        Integer userID = setupResult.get("user_id");
+        Integer projectID = setupResult.get("project_id");
+
+        LoginPage loginPage = new LoginPage();
+        DashboardPage dashboardPage = loginPage.login(usernameToCreate, passwordToCreate);
+        ProjectBoard projectBoard = dashboardPage.goToProjectBoard(projectID);
+        projectBoard.confirm();
+        projectBoard.addTaskToBacklog(taskTitleToAdd);
+
+        //Test
+        projectBoard.openContextMenu();
+        projectBoard.addCommentItem.click();
+        projectBoard.modalSaveBtn.click();
+
+        Assert.assertEquals(projectBoard.errorMsg.text(), "Comment is required");
+
+        //Teardown
+        testTearDown(projectID, userID);
+    }
```

Comment:
Look at the last two tests for comments. They could be moved to separate class and phases `//Setup` and `//Teardown` could be moved to methods `@BeforeMethod` and `@AfterMethod`